src/api-service/__app__/timer_daily/__init__.py

```diff
@@ -7,15 +7,30 @@
 
 import azure.functions as func
 from onefuzztypes.enums import VmState
+from onefuzztypes.events import EventProxyCreated
 
-from ..onefuzzlib.events import get_events
+from ..onefuzzlib.events import get_events, send_event
 from ..onefuzzlib.proxy import Proxy
 from ..onefuzzlib.webhooks import WebhookMessageLog
 from ..onefuzzlib.workers.scalesets import Scaleset
 
 
 def main(mytimer: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa: F841
     for proxy in Proxy.search():
+        if proxy.is_outdated():
+            logging.info("marking proxy as outdated.")
+            proxy.outdated = True
+            proxy.save()
+            if (
```

Comment:
A few comments.

1. `if len(Proxy.search(...)) == 0:` is equivalent `if Proxy.search(...):`
2. You already all of proxy information from line 19.  It requires less back and forth to the database to reuse the data you queried.

Reply:
1. Wouldn't `if len(Proxy.search(...)) == 0` be equivalent to `if **not** Proxy.search()`? 
2. I'm trying to verify that there is not already a 'new' proxy up when this timer goes off - i.e. it will only create a new proxy if one doesn't already exist for this region. If proxy.search() returns a list of proxies, how can I search that list with an equivalent query? I'm thinking of making the following change to avoid querying multiple times: 
`
proxy_list = Proxy.search()
for proxy in proxy_list:
    if proxy.is_outdated():
        logging.info("marking proxy as outdated.")
        proxy.outdated = True
        proxy.save()
        region_list = list(filter(lambda x: (x.region == proxy.region && x.outdated == False), proxy_list)) 
        if not len(region_list):
            logging.info("outdated proxy, creating new one.")`

Reply:
It appears that you have optimized iterations over the proxy list.  I recommend reorganizing this function for clarity, since we expect the number of proxies to be small.  In many deployments, no more than 2.

1. do one pass over the list to mark outdated proxies.
2. do another pass over the list to identify if you need to create a new proxy.

Reply:
![image](https://user-images.githubusercontent.com/74685766/118702033-43477c80-b7c9-11eb-81e5-a30c65b04fcf.png)
Is this more in line with what you're suggesting?